xarray/core/dataset.py

```diff
@@ -10300,3 +10300,18 @@ def resample(
             restore_coord_dims=restore_coord_dims,
             **indexer_kwargs,
         )
+
+    def drop_attrs(self) -> Self:
+        """
+        Removes all attributes from the Dataset and its variables.
+        """
+        # Remove attributes from the dataset
+        self = self.copy()
+
+        self.attrs = {}
+
+        # Remove attributes from each variable in the dataset
+        for var in self.variables:
+            self[var].attrs = {}
```

Comment:
To my understanding this should wipe the attrs of the index_vars of the original object?
https://github.com/pydata/xarray/blob/e8be4bbb961f58ba733852c998f2863f3ff644b1/xarray/core/dataset.py#L1361-L1365

Reply:
Atm an `Index` can't have attrs, at least on `main` testing `ds.indexes['y'].attrs` fails.

I added an explicit test for the coords. Lmk if you have anything else we could do for the index...

Reply:
I wonder if the invariant checks will pass when testing this on a Dataset with a multi-index.

It might be problematic to copy coordinates variables individually when they have a common index and especially when they all wrap *exactly the same index* object in their `._data` attribute (like in the case of a pandas). A safer approach would be to handle the indexed variables with something like this:

```python
new_variables = {}

for idx, idx_vars in self.xindexes.group_by_index():
    # copy each coordinate variable of an index and drop their attrs
    temp_variables = {k: v.copy() for k, v in idx_vars.items()}
    for v in temp_variables.values():
        v.attrs = {}
    # maybe re-wrap the index object in new coordinate variables
    new_variables.update(idx.create_variables(temp_variables))
```

Reply:
Thanks, done. I don't think I've done it elegantly, but I'm a bit out of my depth and so won't try and solve perfectly on this pass...